nautobot/extras/tests/test_jobs.py

```diff
@@ -260,6 +260,33 @@ def test_object_vars(self):
             self.assertEqual(job_result.status, JobResultStatusChoices.STATUS_COMPLETED)
             self.assertEqual({"role": str(d.pk), "roles": [str(d.pk)]}, job_result_data)
 
+    def test_job_data_as_string(self):
+        """
+        Test that job doesn't error when not a dictionary.
+        """
+        with self.settings(JOBS_ROOT=os.path.join(settings.BASE_DIR, "extras/tests/dummy_jobs")):
+            module = "test_object_vars"
+            name = "TestObjectVars"
+            job_class = get_job(f"local/{module}/{name}")
+
+            d = DeviceRole.objects.create(name="role", slug="role")
+
+            # Prepare the job data
+            job_result = JobResult.objects.create(
+                name=job_class.class_path,
+                obj_type=self.job_content_type,
+                user=None,
+                job_id=uuid.uuid4(),
+            )
+            data = "BAD DATA STRING"
+            run_job(data=data, request=None, commit=False, job_result_pk=job_result.pk)
+            job_result.refresh_from_db()
+            job_error_message = job_result.data["null"]["log"][0][4]  # Indexing makes me sad.
```

Comment:
It makes me sad, too. Perhaps extend the comment to make it clear what exactly is being indexed here? The variable name `job_error_message` helps, but it might not be clear for future us.

Reply:
We should be able to change "null" to something more meaningful like "initialization" by setting `job.active_test` when we start initializing the job and jobresult.

Perhaps something like this would be more resilient?

```python
self.assertIn(
    "Data should be a dictionary",
    (
        log[-1]  # actual log message in the logging tuple
        for log in job_result.data["initialization"]["log"]
    ),
)
```